flake.nix

```diff
@@ -24,372 +23,81 @@
-            ] ++ lib.optionals (backend == "llvm")
-              [ "--with-llvm-config=${llvm.dev}/bin/llvm-config" ];
-
-            hardeningDisable = [ "format" ];
-
-            enableParallelBuilding = true;
-
-            passthru = {
-              # run with either of
-              # nix-build -A ghdl-mcode.passthru.tests
-              # nix-build -A ghdl-llvm.passthru.tests
-              tests = {
-                simple = callPackage ./test-simple.nix { inherit backend; };
-              };
-            };
-
-            meta = with lib; {
-              homepage = "https://github.com/ghdl/ghdl";
-              description = "VHDL 2008/93/87 simulator";
-              maintainers = with maintainers; [ lucus16 thoughtpolice ];
-              platforms = platforms.linux;
-              license = licenses.gpl2;
-            };
-          };
+      # Provide some binary packages for selected system types.
+      packages = forAllSystems (system:
+        let
+          pkgs = nixpkgsFor.${system};
+          inherit (pkgs) lib callPackage stdenv fetchgit fetchFromGitHub;
+        in rec {
+          default = yosys;
 
-        abc-verifier = with final;
-          stdenv.mkDerivation rec {
-            pname = "abc-verifier";
-            version = "for-yosys-0.17";
+          ghdl = pkgs.ghdl;
 
+          abc-verifier = pkgs.abc-verifier.overrideAttrs (_: rec {
+            version = "yosys-0.17";
             src = fetchFromGitHub {
               owner = "yosyshq";
               repo = "abc";
-              rev = "09a7e6d";
+              rev = "09a7e6dac739133a927ae7064d319068ab927f90" # == version
+              ;
               hash = "sha256-+1UcYjK2mvhlTHl6lVCcj5q+1D8RUTquHaajSl5NuJg=";
             };
-
-            nativeBuildInputs = [ cmake ];
-            buildInputs = [ readline ];
-
-            installPhase = "mkdir -p $out/bin && mv abc $out/bin";
-
-            # needed by yosys
             passthru.rev = src.rev;
+          });
 
-            meta = with lib; {
-              description =
-                "A tool for squential logic synthesis and formal verification";
-              homepage = "https://people.eecs.berkeley.edu/~alanmi/abc";
-              license = licenses.mit;
-              maintainers = with maintainers; [ thoughtpolice ];
-              mainProgram = "abc";
-              platforms = platforms.unix;
-            };
-          };
-
-        yosys-ghdl = with final;
-          stdenv.mkDerivation {
-            pname = "yosys-ghdl";
-            # This is not the latest commit, but it's the latest that builds with current stable ghdl 2.0.0
-            version = "2022.01.11";
-
-            src = fetchFromGitHub {
-              owner = "ghdl";
-              repo = "ghdl-yosys-plugin";
-              rev = "c9b05e481423c55ffcbb856fd5296701f670808c";
-              sha256 = "sha256-tT2+DXUtbJIBzBUBcyG2sz+3G+dTkciLVIczcRPr0Jw=";
-            };
+          yosys-ghdl = pkgs.yosys-ghdl;
```

Comment:
similar to `ghdl`, this was a copy paste of upstream with no changes, so we can use upstream directly

Reply:
Yes